### ISSUE TITLE
[merp] Some managed assemblies packaged with Mono have zero value timestamp header

### ISSUE THREAD

**Issue by kdubau** [mono/mono#17028]
The following assemblies have a zero-value PE image header timestamp:

* gtk-sharp.dll
* glib-sharp.dll
* pango-sharp.dll

MERP crash report managed frames look like this: 
```
 {
      "is_managed" : "true",
      "guid" : "426A4590-353F-4268-97D5-A3E1EB51B9B8",
      "token" : "0x00000",
      "native_offset" : "0x0",
      "filename" : "gtk-sharp.dll",
      "sizeofimage" : "0x16a000",
      "timestamp" : "0x0",
      "il_offset" : "0x0000c"
   }
```
To symbolicate the frame we look up symbols based on the `filename`, `sizeofimage`, and `timestamp` values.  However, that only works because the assumption is `sizeofimage` and `timestamp` create a _unique_ key for a specific build of the assembly (and it was archived with said key).

According to the [PE image spec](https://docs.microsoft.com/en-us/windows/win32/debug/pe-format?redirectedfrom=MSDN), `sizeofimage` is rounded so it's not always unique on it's own thus the importance of `timestamp`. 

This old comment/thread looks related, https://github.com/dotnet/coreclr/issues/1615#issuecomment-143241535, and probably indicates these assemblies are still built with mcs. 

## Steps to Reproduce

You can use `pedump` to see the header data. For example: `pedump /Library/Frameworks/Mono.framework/Versions/6.4.0/lib/mono/gtk-sharp-2.0/gtk-sharp.dll`

```
kylewhite@kywhi-mbp ~/Workspace/mono (master) $ pedump /Library/Frameworks/Mono.framework/Versions/6.4.0/lib/mono/gtk-sharp-2.0/gtk-sharp.dll | grep "Time stamp:"
	             Time stamp: 0x00000000
```

### Current Behavior

The assembles have a 0x0 timestamp header. 

### Expected Behavior

The assemblies should have a real value, so their symbols can be uniquely archived and retrieved.

**Comment by kdubau** [comment]
Possibly a solution is to build these with roslyn instead of mcs? I am not sure what inherent risk that would introduce.

**Comment by lambdageek** [comment]
@steveisok One for your team, I believe.

**Comment by marek-safar** [comment]
I agree updating the build scripts to use csc is the best way to resolve this 

/cc @directhex

**Comment by Therzok** [comment]
@kdubau definitely doable for mono/gtk-sharp

**Comment by akoeplinger** [comment]
We already did this for gtk-sharp a while back: https://github.com/mono/gtk-sharp/pull/273

It should be in 2019-06 / Mono 6.4.

**Comment by akoeplinger** [comment]
Hmm yeah for some reason I still see the `Time stamp: 0x00000000` even in Mono 6.6.

**Comment by akoeplinger** [comment]
Oh dear 🤦‍♂: https://github.com/mono/bockbuild/blob/fa0a2c10b37b60d6950c7572ad693a7d5bdafed3/packages/gtk-sharp.py#L12

**Comment by directhex** [comment]
Does that build? Last I checked, the gtk-sharp build system does assembly signing in a way csc doesn't understand

**Comment by akoeplinger** [comment]
@directhex yes it does: https://jenkins.mono-project.com/job/build-package-osx-mono-pullrequest/423/ :)

**Comment by akoeplinger** [comment]
Looks like it worked:

```bash
$ for i in gtk-sharp.dll glib-sharp.dll pango-sharp.dll; do pedump /Library/Frameworks/Mono.framework/Versions/6.7.0/lib/mono/gtk-sharp-2.0/$i | grep "Time stamp"; done
	             Time stamp: 0x5d8b7ba0
	             Time stamp: 0x5d8b7b8f
	             Time stamp: 0x5d8b7b91
```

**Comment by akoeplinger** [comment]
The fix landed in 2019-08 and will be in Mono 6.6.0.98. I talked with @marek-safar and it doesn't meet the bar for backporting to 2019-06.

**Comment by vargaz** [comment]
Including timestamps in the binaries make the build process non-deterministic, so we should probably use the assembly guid instead.
